### ISSUE TITLE
(WebCompiler) extend '.clearfix' has no matches

### ISSUE THREAD

**Issue by HugCoder** [madskristensen/WebCompiler#133]
Hi there!

I really like the WebCompiler since it feels more lightweight and user friendly than having grunt and it's dependencies. I'm currently trying to compile the AdminLTE 2.3.1 ( https://github.com/almasaeed2010/AdminLTE ) less files and I just end up with an error/warning on compile, saying
(WebCompiler) extend '.clearfix' has no matches
and
(WebCompiler) extend '.close' has no matches

Upon searching on this I have seen that this was a problem for many on Bootstrap, but I can compile Bootstrap 3.3.5 with WebCompiler, so it's rather weird that this occurs. The reason it happened before was because of a compiler wasn't up to date, but that was some year ago and I'm sure you use something that can handle these things?

I just updated to version BuildWebCompiler.1.6.245 which fixed a few other warnings about undeclared variables that Intellisense didn't pick up. Not sure if that was related or something just kicking in when restarting VS2015 anyway. The above warning seems to prevent me from compiling "AdminLTE.less" at least. Also it only refers to "Line 1" in that file, so there's really nothing of help in the error list.

Any idea on where to look or what to do, or is this an issue or a bug in the WebCompiler?

Regards,
Hug

**Comment by madskristensen** [comment]
I just released version 1.7 of Web Compiler (20 seconds ago) and it has a fix for this. Please check out the new version and reopen the issue if it's not fixed

**Comment by HugCoder** [comment]
Just updated to 1.7. I think it fixed the problem for &:extend(.close); but &:extend(.clearfix); still seems to remain. The weird part is that AdminLTE doesn't even use &:extend(.clearfix); from what I can tell. Only Bootstrap does, and compiling bootstrap.less doesn't produce any errors.
So it still says: (WebCompiler) extend '.clearfix' has no matches

**Comment by madskristensen** [comment]
It give you the warning but it will compile now even with these compiler warnings. That was the fix. Before it didn't actually compile if there were compiler warnings

**Comment by HugCoder** [comment]
I don't get any output though, only:
Process terminated with code 1.
extend ' .clearfix' has no matches
extend ' .close' has no matches
(.close is still present here (Task Runner Explorer) but not error list like before)

no css file is generated. Am I doing something wrong?

**Comment by madskristensen** [comment]
Is the error coming from the NuGet package that compiles on _Build_? If so, you need to update the NuGet package to the latest version as well

**Comment by HugCoder** [comment]
Hmm there we go. I think there was some weird stuff with the update in VS2015 for me. I updated it using the package manager for solution, and it said it was 1.7, but then I checked "extentions and updates..." and there I could find it was still using 1.6 version. Now I updated that and restarted and bam, it still gives same warnings but it says "Compiled with warnings". Of course wished it would say success and what file was compiled but as long as it works I guess it's a step in right direction at least.
Thanks!

**Comment by HugCoder** [comment]
Sorry for having to reopen this but there are still problems regarding this issue. I just did a complete rebuild on my solution and now I get this every time I try to build:
1>  WebCompiler: Begin compiling compilerconfig.json
1>  	Compiled Content/AdminLTE/AdminLTE.css
1>  	Minified Content/AdminLTE/AdminLTE.min.css
1>...\Content\AdminLTE\AdminLTE.less : WebCompiler error 0: extend ' .clearfix' has no matches
1>...\Content\AdminLTE\AdminLTE.less : WebCompiler error 0: extend ' .close' has no matches
1>  WebCompiler: Done compiling compilerconfig.json
========== Build: 0 succeeded, 1 failed, 0 up-to-date, 0 skipped ==========

I can still manually compile the file "with warnings" and the files are compiled, but the solution doesn't build any more. I'm using the latest version available through nuget (1.8.277).

**Comment by jfoegen** [comment]
How did you work around the extend clearfix issue?  Like you said if you do a full rebuild you get the error, but if you just build the file it is fine.

Thanks.

**Comment by mahdibx** [comment]
@madskristensen sorry for replying to a closed issue, but it seems that even though the css is generated now, there are still warnings about that.
`20:29:53 44>..\less\bootstrap-scope.less : WebCompiler error 0: extend ' .btn-xs' has no matches
20:29:53 44>...\less\bootstrap-scope.less : WebCompiler error 0: extend ' .btn-sm' has no matches
20:29:53 44>...\less\bootstrap-scope.less : WebCompiler error 0: extend ' .btn-lg' has no matches`
 And when using the nuget package (latest version), the generation returns a code 1, which makes the build fail.
Could you do something about this?

**Comment by HugCoder** [comment]
Sorry for writing in a closed topic as well. It appears that this issue is now back and kicking. The way I was able to get around it, was adding a few references, like:
'
@import (reference) "../bootstrap-less/mixins.less";
@import (reference) "../bootstrap-less/variables.less";
@import (reference) "mixins.less";
@import (reference) "variables.less";
'

but I tried to add it pretty much everywhere now, and I get both an Error saying:
'
extend '.clearfix' has no matches
extend '.close' has no matches
'
and a Warning, saying the same thing. This blocks the build. I'm using Web Compiler 1.11.326.
